### PR TITLE
Specify include_type_name in HTTP monitoring.

### PR DESCRIPTION
**Pull request by jtibshirani**
In 6.7, `include_type_name` must be specified in order to avoid deprecation
warnings. Here we make sure to use `include_type_name` when setting up templates
for the monitoring HTTP exporter.

Addresses #37442.

### CODE REVIEW

**Review by ycombinator — APPROVED** [review]
LGTM.

File: x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/exporter/http/TemplateHttpResource.java [hunk 1]
```diff
@@ -31,6 +36,15 @@
 
     private static final Logger logger = LogManager.getLogger(TemplateHttpResource.class);
 
+    private static final Map<String, String> PARAMETERS;
+
+    static {
+        Map<String, String> parameters = new HashMap<>(
+            PublishableHttpResource.RESOURCE_VERSION_PARAMETERS);
+        parameters.put(INCLUDE_TYPE_NAME_PARAMETER, "true");
+        PARAMETERS = Collections.unmodifiableMap(parameters);
```
Comment (jakelandis) [inline x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/exporter/http/TemplateHttpResource.java]: nit: I think you can use https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/common/collect/MapBuilder.java#L87 to shorten this up abit.

**Review by jakelandis — APPROVED** [review]
LGTM

File: x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/exporter/http/TemplateHttpResource.java (hunk 1 above)
Comment (jtibshirani) [reply x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/exporter/http/TemplateHttpResource.java]: Nice, I didn't realize we had that.

**Comment by jtibshirani** [comment]
@jakelandis @ycombinator thank you for the reviews! After this merges, it would be good to test HTTP monitoring on 6.7 to make sure there are no deprecation warnings.

**Pull request merged**